### PR TITLE
Update aws-sam-deployments.md

### PR DESCRIPTION
**Pull request by roshnisarangadharan**
https://harness.atlassian.net/browse/CDS-95828

### CODE REVIEW

File: docs/continuous-delivery/deploy-srv-diff-platforms/aws/aws-sam-deployments.md [hunk 1]
```diff
@@ -250,7 +250,9 @@ These steps are described in detail below.
 
 Harness automatically generates a containerized step group containing the steps needed for the SAM build and deploy.
 
-When the step group setting **Enable container based execution** is enabled, the step group is containerized. When it is disabled, the Harness Delegate performs all tasks on its host system.
+When the step group setting **Enable container based execution** is enabled, the step group is containerized. SAM runs on containerized step groups only. You need not install SAM CLI on Harness Delegate as SAM steps are baked in with the SAM CLI. Hence, the SAM steps run in the containers that re brought up inside the pod within the Kubernetes cluster configured in the step group configuration.
```
Comment (piyush-bhuwalka) [inline docs/continuous-delivery/deploy-srv-diff-platforms/aws/aws-sam-deployments.md:253]: " as SAM steps are baked" -> " as Harness provided SAM step images are baked"

Comment (piyush-bhuwalka) [reply docs/continuous-delivery/deploy-srv-diff-platforms/aws/aws-sam-deployments.md:253]: "Hence, the SAM steps run" -> "SAM steps run"

Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/aws/aws-sam-deployments.md:253]:
```suggestion
When the step group setting **Enable container based execution** is enabled, the step group is containerized. SAM runs on containerized step groups only. You need not install SAM CLI on Harness Delegate as Harness provided SAM step images are baked within the SAM CLI. SAM steps run in the containers that are brought up inside the pod within the Kubernetes cluster configured in the step group configuration. 
```

**Review by piyush-bhuwalka — APPROVED** [review]

File: docs/continuous-delivery/deploy-srv-diff-platforms/aws/aws-sam-deployments.md [hunk 2]
```diff
@@ -250,7 +250,9 @@ These steps are described in detail below.
 
 Harness automatically generates a containerized step group containing the steps needed for the SAM build and deploy.
 
-When the step group setting **Enable container based execution** is enabled, the step group is containerized. When it is disabled, the Harness Delegate performs all tasks on its host system.
+When the step group setting **Enable container based execution** is enabled, the step group is containerized. SAM runs on containerized step groups only. You need not install SAM CLI on Harness Delegate as Harness provided SAM step images are baked within the SAM CLI. SAM steps run in the containers that are brought up inside the pod within the Kubernetes cluster configured in the step group configuration. 
+
+When the **Enable container based execution** setting is disabled, the Harness Delegate performs all tasks on its host system.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/aws/aws-sam-deployments.md:255]:
```suggestion
When the **Enable container based execution** setting is disabled, Harness Delegate performs all tasks on its host system.
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/aws/aws-sam-deployments.md [hunk 3]
```diff
@@ -250,7 +250,9 @@ These steps are described in detail below.
 
 Harness automatically generates a containerized step group containing the steps needed for the SAM build and deploy.
 
-When the step group setting **Enable container based execution** is enabled, the step group is containerized. When it is disabled, the Harness Delegate performs all tasks on its host system.
+When the step group setting **Enable container based execution** is enabled, the step group is containerized. SAM runs on containerized step groups only. You need not install SAM CLI on Harness Delegate as Harness provided SAM step images are baked within the SAM CLI. SAM steps run in the containers that are brought up inside the pod within the Kubernetes cluster configured in the step group configuration. 
+
+When the **Enable container based execution** setting is disabled, Harness Delegate performs all tasks on its host system.
```
Comment (ivan-83) [inline docs/continuous-delivery/deploy-srv-diff-platforms/aws/aws-sam-deployments.md:255]: I think the point here is to mention that, if **Enable container based execution** setting is disabled, there is no sense in using the SAM build/deploy steps within the step group. The delegate can't do SAM deployment. 
cc: @piyush-bhuwalka

Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/aws/aws-sam-deployments.md:255]:
```suggestion
When the **Enable container based execution** setting is disabled, Harness Delegate cannot perform the SAM deployment.
```

**Review by ivan-83 — APPROVED** [review]

**Review by piyush-bhuwalka — APPROVED** [review]

**Pull request merged**